### ISSUE TITLE
Custom rules integration to IntelliJ IDEA plugin?

### ISSUE THREAD

**Issue by mtiidla** [arturbosch/detekt#1864]
I have created a custom detekt rule and integrated it into the CI process. I am also using the detekt IntelliJ plugin for Android Studio which works like a charm highlighting the issues based on the detekt.yml configuration file.

I have tried to add my ruleset to that file with the following config:
```
my_project_detekt_rules:
  NoKoinExplicitReturnType:
    active: true
```
I have also selected the .yml file in the IDE plugin as the configuration file and can confirm it registers changes in the yml file (IDE won't show warnings on a issue if I disable it in the .yml file)

However, the IDE does not seem to pick up my custom rule to highlight the issues, is that even supported by the plugin, or how should I approach that?

PS. if I turn this "active: true" flag to false, it it still active. Do I need to extend some other rule?
`class NoKoinExplicitReturnType : Rule() { ... }`

Using version 1.0.1

**Comment by schalkms** [comment]
Thanks for reporting this behavior.
Does the **active=false** work in your case if you use detekt without the IntelliJ plugin?
Extending the `Rule` class is fine. If the custom rule works without the IntelliJ plugin, this issue should be moved to the corresponding plugin repo.

**Comment by mtiidla** [comment]
I tried now with plain gradle task, managed to register my plugin nicely using "--plugins /path/to/jar/", but trying to set "active:false" in the default-detekt-config.yml does not disable it. (Just in case, I also tried setting "active:false" for  FunctionNaming and that was properly disabled. Maybe I am missing the correct syntax to register my custom rule in the default-detekt-config.yml? Right now I just try to match it by using the same name for the Rule class and the .yml property.

**Comment by schalkms** [comment]
How does your rule look like?
Have you passed the `Config` object to the `Rule` base class like in the [sample project](https://arturbosch.github.io/detekt/extensions.html)?

```kotlin
class YourCustomRule(config: Config) : 
    Rule("YourCustomRule", Severity.Defect, config) {
```

Could you please also try to use on of the default Gradle configuration from the [documentation](https://arturbosch.github.io/detekt/groovydsl.html)?

**Comment by mtiidla** [comment]
Passing the config to the Rule superclass did the trick. I followed https://github.com/arturbosch/detekt/blob/master/detekt-sample-extensions/src/main/kotlin/io/gitlab/arturbosch/detekt/sample/extensions/rules/TooManyFunctions.kt sample and missed it as it is unfortunately not shown there.

Final solution:
`class NoKoinExplicitReturnType(config: Config) : Rule(config) { ... }`

Funny enough, now I am quite confused:
1) Pass config to Rule constructor, need to define active: true in .yml config, otherwise it is not enabled.
2) Don't pass config to Rule constructor, no need to even mention the rule in .yml config, already enabled. (Looking at the detekt-api source code it doesn't make sense).

PS. the example code block in your reply is outdated, Rule constructor no longer accepts rule name nor severity.

**Comment by schalkms** [comment]
I’m glad that you could fix this issue. We’re sorry for confusing you.
The documentation is unfortunately outdated. I’ll update it.
Thank you for your feedback! We’ll definitely incorporate it in an update.

@arturbosch why do we even provide two different sample rules, whereby one passes a config object while the other does not?

**Comment by mtiidla** [comment]
Thank you for the quick response and I'm glad the documentation will get a bit more love :) Close as you wish.
